crates/database/src/trades.rs

```diff
@@ -43,14 +43,20 @@ LEFT OUTER JOIN LATERAL (
     LIMIT 1
 ) AS settlement ON true
 JOIN orders o
-ON o.uid = t.order_uid
-WHERE
-    o.uid IS NOT null
-AND
-    ($1 IS NULL OR o.owner = $1)
-AND
-    ($2 IS NULL OR o.uid = $2)
-    "#;
+ON o.uid = t.order_uid"#;
+    const QUERY: &str = const_format::concatcp!(
+        COMMON_QUERY,
+        " WHERE o.uid IS NOT null ",
+        " AND ($1 IS NULL OR o.owner = $1)",
+        " AND ($2 IS NULL OR o.uid = $2)",
+        "UNION ALL",
```

Comment:
This union all improves the performance of my unit test by a factor of 10. 42ms vs 3.7ms.
I don't understand it, but its similar to the performance that we have seen here: https://github.com/cowprotocol/services/pull/610